packages/std/src/ibc.rs

```diff
@@ -115,6 +115,7 @@ impl From<IbcTimeoutBlock> for IbcTimeout {
 /// IbcChannel defines all information on a channel.
 /// This is generally used in the hand-shake process, but can be queried directly.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[non_exhaustive]
```

Comment:
There is no constructor (::new) for this, just the `mock_ibc_channel`, right?)

I guess for the contract tests we don't need anything else, but it seems very restrictive to not have any non-test constructor. I can see the need for this if I build out IBC testing support into multi-test (in cosmwasm-plus)

Reply:
Same as `IbcPacket` - I didn't find anywhere where we're actually constructing these outside of `cosmwasm-std`, so I didn't bother. I also figured we can add that without breaking anything, so it could be added easily before/after 1.0.

I'm happy to add some constructors now though. `IbcPacket` and IbcChannel` both? Just a `new` thing that sets every field? It doesn't look like these things warrant a builder pattern type thing.

Reply:
Then again, a `new` that sets everything is how we make it harder to evolve later on.

Reply:
Yes, just a simple `new` that sets key fields and leaves other as default. (Or sets all)

I think I may need a way to do this with some future multi-test changes, and would like at least some way to construct those without waiting for a cosmwasm patch release.